processor/processor.go

```diff
@@ -797,6 +798,106 @@ func (proc *HandleT) updateSourceEventStatsDetailed(event types.SingularEventT,
+			for _, transformerEvent := range tStatus.eventsToTransform {
+				if transformerEvent.Metadata.MessageID != "" {
+					reportedMessageIDs[transformerEvent.Metadata.MessageID] = struct{}{}
+					singularEventWithReceivedAt := tStatus.eventsByMessageID[transformerEvent.Metadata.MessageID]
+					if _, ok := eventBeforeMap[transformerEvent.Metadata.MessageID]; !ok {
+						eventBeforeMap[transformerEvent.Metadata.MessageID] = getEvent(singularEventWithReceivedAt.SingularEvent, singularEventWithReceivedAt.ReceivedAt)
+					}
+
+					if _, ok := eventAfterMap[transformerEvent.Metadata.MessageID]; !ok {
+						eventAfterMap[transformerEvent.Metadata.MessageID] = getEvent(transformerEvent.Message, time.Now())
+					} else {
+						payloadArr := eventAfterMap[transformerEvent.Metadata.MessageID]["payload"].([]map[string]interface{})
+						payloadArr = append(payloadArr, transformerEvent.Message)
+						eventAfterMap[transformerEvent.Metadata.MessageID]["payload"] = payloadArr
+					}
+				}
+			}
+			for k := range eventBeforeMap {
+				transformationdebugger.RecordTransformationStatus(transformation.VersionID, tStatus.sourceID, tStatus.destID, eventBeforeMap[k], eventAfterMap[k], false)
+			}
+
+			for _, failedEvent := range tStatus.response.FailedEvents {
+				if len(failedEvent.Metadata.MessageIDs) > 0 {
+					for _, msgID := range failedEvent.Metadata.MessageIDs {
+						reportedMessageIDs[msgID] = struct{}{}
+						singularEventWithReceivedAt := tStatus.eventsByMessageID[msgID]
+						eventBefore := getEvent(singularEventWithReceivedAt.SingularEvent, singularEventWithReceivedAt.ReceivedAt)
+						eventAfter := map[string]interface{}{
+							"error":      failedEvent.Error,
+							"statusCode": failedEvent.StatusCode,
+							"isDropped":  false,
+						}
+						transformationdebugger.RecordTransformationStatus(transformation.VersionID, tStatus.sourceID, tStatus.destID, eventBefore, eventAfter, true)
+					}
+				} else if failedEvent.Metadata.MessageID != "" {
+					reportedMessageIDs[failedEvent.Metadata.MessageID] = struct{}{}
+					singularEventWithReceivedAt := tStatus.eventsByMessageID[failedEvent.Metadata.MessageID]
+					eventBefore := getEvent(singularEventWithReceivedAt.SingularEvent, singularEventWithReceivedAt.ReceivedAt)
+					eventAfter := map[string]interface{}{
+						"error":      failedEvent.Error,
+						"statusCode": failedEvent.StatusCode,
+						"isDropped":  false,
+					}
+					transformationdebugger.RecordTransformationStatus(transformation.VersionID, tStatus.sourceID, tStatus.destID, eventBefore, eventAfter, true)
+				}
+			}
+
+			for msgID := range tStatus.uniqueMessageIds {
+				if _, ok := reportedMessageIDs[msgID]; !ok {
+					singularEventWithReceivedAt := tStatus.eventsByMessageID[msgID]
+					eventBefore := getEvent(singularEventWithReceivedAt.SingularEvent, singularEventWithReceivedAt.ReceivedAt)
+					eventAfter := map[string]interface{}{
+						"isDropped": true,
```

Comment:
change `isDropped` to `unprocessed` or something. we are generally referring dropped to indicate transfomer errors in discussions kada?